pkg/operator/operator.go

```diff
@@ -743,3 +788,82 @@ func (o *Operator) Config(ctx context.Context, key string) (*manifests.Config, e
 
 	return c, nil
 }
+
+// Upgradeable verifies whether the operator can be upgraded or not. It returns
+// the ConditionStatus with optional reason and message.
+// To set this status, it will verify that in HA topology, workloads with
+// persistent storage are correctly spread across multiple nodes. If it isn't
+// it will try to rebalance the workloads.
+func (o *Operator) Upgradeable(ctx context.Context) (configv1.ConditionStatus, string, string, error) {
+	if !o.lastKnowInfrastructureConfig.HighlyAvailableInfrastructure() {
+		return configv1.ConditionTrue, "", "", nil
+	}
+
+	var (
+		messages           []string
+		workloadRebalanced bool
+	)
+	for _, workload := range o.workloadsToRebalance() {
+		balanced, err := o.rebalancer.WorkloadCorrectlyBalanced(ctx, &workload)
+		if err != nil {
+			return configv1.ConditionUnknown, "", "", err
+		}
+
+		if balanced {
+			continue
+		}
+
+		workloadRebalanced, err := o.rebalancer.RebalanceWorkloads(ctx, &workload)
+		if err != nil {
+			return configv1.ConditionUnknown, "", "", err
+		}
+
+		if !workloadRebalanced {
+			messages = append(messages, fmt.Sprintf("Highly-available workload %s/%s with persistent storage has a single point of failure.", workload.Namespace, workload.Name))
+		}
+	}
+
+	if len(messages) > 0 {
+		msg := "Manual intervention is needed to upgrade to the next minor version. Please refer to the following documentation to fix this issue: https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md."
```

Comment:
the runbook is named after the alert so we'd have to modify the alert too which might be more work than needed.

Reply:
My plan is to remove the alert and the runbook once this is merged so I will update the message to explain what needs to be done by the administrator instead of linking to the runbook.